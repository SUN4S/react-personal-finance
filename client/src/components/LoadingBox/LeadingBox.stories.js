import { LoadingBox } from "./LoadingBox";

export default {
  title: "Loading Box",
  component: LoadingBox,
};

export const Default_Light = () => (
  <html data-theme="light">
    <div
      style={{
        backgroundColor: "var(--sidebar-bg)",
      }}
    >
      <LoadingBox size="sm" />
      <LoadingBox size="md" />
      <LoadingBox size="lg" />
      <LoadingBox size="xl" />
    </div>
  </html>
);

export const Default_Dark = () => (
  <html data-theme="dark">
    <div
      style={{
        backgroundColor: "var(--sidebar-bg)",
      }}
    >
      <LoadingBox size="sm" />
      <LoadingBox size="md" />
      <LoadingBox size="lg" />
      <LoadingBox size="xl" />
    </div>
  </html>
);
