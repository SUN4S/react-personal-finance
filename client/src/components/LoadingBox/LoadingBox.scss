.loadingContainer {
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 12px;

  .loadingSpinner {
    border: 2px solid var(--primary);
    border-bottom: 2px solid var(--page-bg);
    border-radius: 50%;
    animation: spin 2s linear infinite;

    &.sm {
      height: 12px;
      width: 12px;
    }

    &.md {
      height: 24px;
      width: 24px;
    }

    &.lg {
      height: 48px;
      width: 48px;
    }

    &.xl {
      height: 96px;
      width: 96px;
    }

    @keyframes spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
  }
}
