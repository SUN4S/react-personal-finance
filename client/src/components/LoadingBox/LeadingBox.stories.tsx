import { Meta, Story } from "@storybook/react/types-6-0";

import { LoadingBox } from "./LoadingBox";

interface TemplateProps {
  theme: string;
  size: string;
}

export default {
  title: "Components/Loading Box",
  component: LoadingBox,
  argTypes: {
    size: {
      options: ["sm", "md", "lg", "xl"],
      control: { type: "radio" },
    },
  },
} as Meta;

const Template: Story<TemplateProps> = (args) => {
  return (
    <html data-theme={args.theme}>
      <div
        style={{
          backgroundColor: "var(--sidebar-bg)",
        }}
      >
        <LoadingBox size={args.size} />
      </div>
    </html>
  );
};

export const Default_Light = Template.bind({});
Default_Light.args = {
  theme: "light",
  size: "sm",
};

export const Default_Dark = Template.bind({});
Default_Dark.args = {
  theme: "dark",
  size: "sm",
};
