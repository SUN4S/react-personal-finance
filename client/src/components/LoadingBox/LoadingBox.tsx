import "./LoadingBox.scss";

export interface LoadingProps {
  size: "sm" | "md" | "lg" | "xl";
}
export const LoadingBox = (props: { size: string }) => {
  return (
    <div className="loadingContainer" data-testid="loadingBox">
      <div className={`loadingSpinner ${props.size}`} data-testid="loadingSpinner"></div>
    </div>
  );
};
