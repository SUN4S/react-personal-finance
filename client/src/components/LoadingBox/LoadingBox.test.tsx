import { cleanup, fireEvent, render } from "@testing-library/react";

import { LoadingBox } from "./LoadingBox";
import renderer from "react-test-renderer";

describe("Loading Box", () => {
  afterEach(cleanup);

  it("Renders", () => {
    const { getByTestId } = render(<LoadingBox size="sm" />);
    expect(getByTestId("loadingBox")).toBeTruthy();
  });

  it("Sets Size Small", () => {
    const { getByTestId } = render(<LoadingBox size="sm" />);
    expect(getByTestId("loadingSpinner")).toHaveClass("sm");
  });

  it("Sets Size large", () => {
    const { getByTestId } = render(<LoadingBox size="lg" />);
    expect(getByTestId("loadingSpinner")).toHaveClass("lg");
  });

  it("Matches Snapshot", () => {
    const tree = renderer.create(<LoadingBox size="lg" />);
    expect(tree).toMatchSnapshot();
  });
});
