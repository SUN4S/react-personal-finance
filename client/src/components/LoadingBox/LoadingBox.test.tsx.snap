// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Box Matches Snapshot 1`] = `
<div
  className="loadingContainer"
  data-testid="loadingBox"
>
  <div
    className="loadingSpinner lg"
    data-testid="loadingSpinner"
  />
</div>
`;
